src/pal/pal_netbsd.h

```diff
@@ -26,7 +28,44 @@ namespace snmalloc
      * As NetBSD does not have the getentropy call, get_entropy64 will
      * currently fallback to C++ libraries std::random_device.
      */
-    static constexpr uint64_t pal_features = PALBSD_Aligned::pal_features;
+    static constexpr uint64_t pal_features =
+      PALBSD_Aligned::pal_features | Entropy;
+
+    /**
+     * Temporary solution while waiting getrandom support for the next release
+     * random_device seems unimplemented in clang for this platform
+     */
+    static uint64_t get_entropy64()
+    {
+      uint64_t result;
+      auto flags = O_RDONLY | O_CLOEXEC;
+      auto fd = open("/dev/urandom", flags, 0);
```

Comment:
This looks like it's the same as the Linux version - can we factor it out into a function that other operating systems that provide a /dev/[u]random can use?

Reply:
almost :-) but I favored linux version regardless.